CelesteNet.Shared/CelesteNetClientOptions.cs

```diff
@@ -6,9 +6,12 @@ public class CelesteNetClientOptions {
         public bool AvatarsDisabled = false;
         public ulong ClientID;
         public uint InstanceID;
-        public CelesteNetSupportedClientFeatures SupportedClientFeatures;
+        // Due to how these are sent as packets, this needs to be a ulong.
```

Comment:
It doesn't matter much in the end, but I don't really see why this change needed to be made 🤔 

The Client sends these to the Server during the Handshake via Reflection (so these aren't "sent as packets") and even with the enum used directly this should still be treated as a TypeCode.UInt64 either way. Where did this run into issues?

![image](https://github.com/user-attachments/assets/2ba5a766-e268-47e8-9365-fb5344f24824)

At least all of this worked for me in a REPL and that's all I checked for now, not actually gonna try to revert the change 😅

Reply:
Oh wait I see it now... `field.GetValue(options)` will literally write "LocateCommand" into the headers... and then `UInt64.Parse("LocateCommand")` is the part that would fail on the server... :hmm:

So technically this is a limitation in how the Client writes those values, because of the Reflection...

Technically Handshake.cs in the client could do something truly comical like
```cs
string value = (!field.FieldType.IsEnum ? field.GetValue(options) : field.FieldType.GetEnumValuesAsUnderlyingType().GetValue(Array.IndexOf(field.FieldType.GetEnumValues(), field.GetValue(options)))).ToString();
```
or I have no clue how to get an enum's underlying value with Reflection